runtime/lua/vim/uri.lua

```diff
@@ -78,11 +78,19 @@ end
 
 local URI_SCHEME_PATTERN = '^([a-zA-Z]+[a-zA-Z0-9.+-]*):.*'
 local WINDOWS_URI_SCHEME_PATTERN = '^([a-zA-Z]+[a-zA-Z0-9.+-]*):[a-zA-Z]:.*'
+--https://url.spec.whatwg.org/#special-scheme
+local SPECIAL_SCHEME  = {'ftp', 'file', 'http', 'https', 'ws', 'wss'}
 
 --- Get a URI from a bufnr
 ---@param bufnr number
 ---@return string URI
 local function uri_from_bufnr(bufnr)
+  local relative_name = vim.fn.bufname(bufnr)
+  local maybe_scheme = relative_name:match(URI_SCHEME_PATTERN)
```

Comment:
I haven't followed the thread, so sorry if this has been covered, but why can't `relative_name` be `fname` here?

Reply:
`nvim_buf_get_name` always returns the full path. When you create a buffer with a name `deno:some/file`, the current directory will be prepended to it. With `bufname()` it prefers returning the short name (`deno:some/file`), but it's generally not 100% reliable.

Reply:
Then this is all a nasty hack. `nvim_buf_get_name` is doing exactly the right thing. We shouldn't be relying on the cwd the parse the correct uri.

deno should be using something like `deno://som/file`?

Reply:
`deno:some/file` is technically compliant with the spec, vim just treats it like a normal path.

Two potential solutions we have right now is to start complying with the URI spec, or translate paths to and from `deno://`

Reply:
How feasible do we think it is to comply with the URI spec? I think that would be preferable if it doesn't break too much.

Reply:
It should be straightforward, all checks for `://` have to be replaced with `:`, I think it's just in few places. But yeah, the question is how much of the user stuff would it break.